File: azure-spark-maven-plugin/src/main/java/com/microsoft/azure/MyMojo.java
package com.microsoft.azure;

/**
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 */

import org.apache.maven.plugin.AbstractMojo;
import org.apache.maven.plugin.MojoExecutionException;
import org.apache.maven.plugins.annotations.Mojo;

@Mojo(name = "sayHi")
public class MyMojo
    extends AbstractMojo
{
    public void execute()
        throws MojoExecutionException
    {
        getLog().info("Hello world!");
    }
}

File: azure-spark-maven-plugin/src/main/java/com/microsoft/azure/MyMojo.kt
package com.microsoft.azure

/**
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 */

import com.microsoft.azure.maven.AbstractAzureMojo
import org.apache.maven.plugins.annotations.Mojo

@Mojo(name = "sayHi")
class MyMojo : AbstractAzureMojo() {
    override fun doExecute() {
        log.info("Hello world from kt!")
    }
}
